src/Components/Admins/Profile/index.jsx

```diff
@@ -0,0 +1,231 @@
+import Loader from 'Components/Shared/Loader';
+import Button from 'Components/Shared/Button';
+import { Input } from 'Components/Shared/Inputs';
+import ConfirmModal from 'Components/Shared/ConfirmModal';
+import ResponseModal from 'Components/Shared/ResponseModal';
+import { getAdmins, editAdmin, deleteAdmin } from 'Redux/Admins/thunks';
+import { handleDisplayToast } from 'Redux/Shared/ResponseToast/actions';
+import React, { useState, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+import { useForm } from 'react-hook-form';
+import adminSchema from 'Validations/admin';
+import { joiResolver } from '@hookform/resolvers/joi';
+function AdminProfile() {
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const [disableEdit, setDisableEdit] = useState(true);
+  const [action, setAction] = useState('');
+  const [showConfirmModal, setShowConfirmModal] = useState(false);
+  const { show, message, state } = useSelector((state) => state.toast);
+  const loading = useSelector((state) => state.admins.isPending);
+  const defaultAdmin = useSelector((state) => state.admins.defaultAdmin);
+
+  const {
+    register,
+    handleSubmit,
+    setValue,
+    clearErrors,
+    formState: { errors }
+  } = useForm({
+    mode: 'onChange',
+    resolver: joiResolver(adminSchema),
+    defaultValues: {
+      firstName: defaultAdmin.firstName,
+      lastName: defaultAdmin.lastName,
+      dni: defaultAdmin.dni,
+      phone: defaultAdmin.phone,
+      email: defaultAdmin.email,
+      city: defaultAdmin.city,
+      password: defaultAdmin.password
+    }
+  });
+  useEffect(() => {
+    getAdmins(dispatch);
+  }, [dispatch]);
+
+  useEffect(() => {
+    setValue('firstName', defaultAdmin.firstName);
+    setValue('lastName', defaultAdmin.lastName);
+    setValue('dni', defaultAdmin.dni);
+    setValue('phone', defaultAdmin.phone);
+    setValue('email', defaultAdmin.email);
+    setValue('city', defaultAdmin.city);
+    setValue('password', defaultAdmin.password);
+  }, [defaultAdmin]);
+
+  const onSubmit = (data) => {
+    editAdmin(dispatch, defaultAdmin._id, data);
+    setShowConfirmModal(false);
+  };
+
+  const handleDeleteAdmin = () => {
+    deleteAdmin(dispatch, defaultAdmin._id);
+    setShowConfirmModal(false);
+    history.push('/');
+  };
+  const handleAction = (action) => {
+    setShowConfirmModal(true);
+    setAction(action);
+  };
+
+  const handleClose = () => {
+    setDisableEdit(true);
+    setValue('firstName', defaultAdmin.firstName);
+    setValue('lastName', defaultAdmin.lastName);
+    setValue('dni', defaultAdmin.dni);
+    setValue('phone', defaultAdmin.phone);
+    setValue('email', defaultAdmin.email);
+    setValue('city', defaultAdmin.city);
+    setValue('password', defaultAdmin.password);
+    clearErrors();
+  };
+
+  return (
+    <div className={styles.form}>
+      {defaultAdmin == {} && <p>There are no admins</p>}
```

Comment:
I think that the empty object couldn't be evaluated this way, I found this way that works!

```suggestion
      {Object.keys(defaultAdmin).length === 0 && <p>There are no admins</p>}
```

Reply:
awesome, thank you!